mmv1/third_party/terraform/README.md

```diff
@@ -242,4 +369,44 @@ end in `.go.erb`.
 <% end -%>
 ```
 
+Otherwise, tests using a beta feature are written exactly the same as tests
+using a GA one. Normally to use the beta provider, it's necessary to specify
+`provider = google-beta`, as Terraform maps any resources prefixed with
+`google_` to the `google` provider by default. However, inside the test
+framework, the `google-beta` provider has been aliased as the `google` provider
+and that is not necessary.
+
+Note: You _may_ use version guards to test different configurations between the
+GA and beta provider tests, but it's strongly recommended that you write
+different test cases instead, even if they're slightly duplicative.
+
 #### Promote a beta feature
+
+"Promoting" a beta feature- making it available in the GA `google` provider when
+the underlying feature or service has gone GA- requires removing the version
+guards placed previously, so that the previously beta-only code is generated in
+the `google` provider as well.
+
+For all promotions, ensure that you remove the guards in:
+
+* The documentation for the resource or field
+* The test(s) for the resource or field.
+
+For whole resource promotions, you'll generally only need to remove the file-level
+guards.
+
+For field promotions ensure that you remove the guards in:
+
+* The Resource schema
+* The Resource CRUD methods (for top level fields)
+* The Resource Expanders and Flatteners (for nested fields)
+
+When writing a changelog entry for a promotion, write it as if it was a new
```

Comment:
Is this also the rule for field promotion? I thought we wrote "promoted `field` in `resource` to GA"